src/routing.js

```diff
@@ -55,6 +60,29 @@ class Graph {
    */
   bfs(start) {
     // !!! IMPLEMENT ME
+    for (let v of this.vertexes) {
+      v.color = 'white';
+      v.parent = null;
+    }
+
+    start.color = 'gray';
+
+    const queue = [];
+    queue.push(start);
+    while (queue.length > 0) {
+      const vertex = queue[0];
+
+      for (let edge of vertex.edges) {
+        if (edge.destination.color === 'white'){
+          edge.destination.color = 'gray';
+          edge.destination.parent = vertex;
+          queue.push(edge.destination);
+        }
+      }
+      // console.log(vertex.parent);
+      vertex.color = 'black';
+      queue.shift();
+    }
```

Comment:
Good clean loop ✨ 

Don't forget to get rid of your `console.log`s when you don't need them anymore